### PR TITLE
Add experimental setting option for redesigned pages

### LINKED ISSUE #23554: Confirmation redesign - add preference flag for re-designed signature request pages
User should be able to access redesigned pages if preference is enabled.

[Figma design](https://www.figma.com/file/7kYok9B46ohR6h47cksRAz/Confirmation-redesign-V1?type=design&node-id=1720-12461&mode=design&t=Dayzg9Q0KNS1Ax51-0) (still needs copy)

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**
Add preference for re-designed flag in experimental setting.

## **Related issues**

Fixes: MetaMask/metamask-extension#23554

## **Manual testing steps**

1. Enable flag `ENABLE_CONFIRMATION_REDESIGN`
2. Go to experimental settings
3. Enable re-designed confirmations in settings
4. Trigger signature request in test dapp, it should be visible in new designs

## **Screenshots/Recordings**
Screenshot 2024-04-16 at 7 22 09 PM

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/pages/settings/experimental-tab/experimental-tab.component.js [hunk 1]
```diff
@@ -109,6 +111,37 @@ export default class ExperimentalTab extends PureComponent {
...
   }
 
+  renderToggleRedesignedConfirmations() {
+    const { t } = this.context;
+    const { redesignedConfirmations, setRedesignedConfirmationsEnabled } =
+      this.props;
+
+    return (
+      <Box
+        ref={this.settingsRefs[0]}
```
Comment (pedronfigueiredo) [inline ui/pages/settings/experimental-tab/experimental-tab.component.js:121]: Can we add this toggle to the settings search?

File: ui/pages/settings/experimental-tab/experimental-tab.test.js [hunk 2]
```diff
@@ -79,4 +79,24 @@ describe('ExperimentalTab', () => {
       expect(setPetnamesEnabled).toHaveBeenCalledWith(false);
     });
   });
+
+  it('should enable redesigned confirmations', async () => {
```
Comment (pedronfigueiredo) [inline ui/pages/settings/experimental-tab/experimental-tab.test.js:83]: I wonder if we should add an e2e test that opens a bunch of signatures from the test dApp and checks which version is displayed based on the toggle state

Comment (cryptotavares) [reply ui/pages/settings/experimental-tab/experimental-tab.test.js:83]: We can add that validation on existing e2e tests no?
What we could do as well is adding an e2e test with the toggle and env var on for all different confirmations. And then as soon as we migrate a new confirmation type, we would need to update that test.

File: ui/pages/confirmations/hooks/useCurrentConfirmation.ts [hunk 3]
```diff
@@ -27,9 +28,15 @@ const useCurrentConfirmation = () => {
...
+    getRedesignedConfirmationsEnabled,
+  );
 
   useEffect(() => {
-    if (!process.env.ENABLE_CONFIRMATION_REDESIGN) {
+    if (
+      !process.env.ENABLE_CONFIRMATION_REDESIGN ||
+      !redesignedConfirmationsEnabled
+    ) {
       return;
```
Comment (cryptotavares) [inline ui/pages/confirmations/hooks/useCurrentConfirmation.ts:40]: [Not related to this task] Eventually we would want to refactor/simplify this file. Branching growing a lot on the useEffect.

File: ui/pages/settings/experimental-tab/experimental-tab.test.js (hunk 2 above)
Comment (pedronfigueiredo) [reply ui/pages/settings/experimental-tab/experimental-tab.test.js:83]: Exactly, that's what I meant with my suggestion.

Comment (jpuri) [reply ui/pages/settings/experimental-tab/experimental-tab.test.js:83]: This is sort of temporary feature and will go away as we move to new designs, this I am bit reluctant to add e2e for it.

File: ui/pages/confirmations/hooks/useCurrentConfirmation.ts (hunk 3 above)
Comment (jpuri) [reply ui/pages/confirmations/hooks/useCurrentConfirmation.ts:40]: As we move all confirmations to new designs this branching will actually simplify.

File: ui/pages/settings/experimental-tab/experimental-tab.component.js (hunk 1 above)
Comment (jpuri) [reply ui/pages/settings/experimental-tab/experimental-tab.component.js:121]: I updated the PR to address this.

**Review by pedronfigueiredo — APPROVED** [review]

**Review by cryptotavares — APPROVED** [review]

**Pull request merged**